engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/ConfluentAvroMessageFormatter.scala

```diff
@@ -1,15 +1,21 @@
 package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.formatter
 
-import java.io.PrintStream
+import java.io.ByteArrayOutputStream
 import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
 
-import io.confluent.kafka.schemaregistry.avro.AvroSchemaUtils
+import io.circe.generic.semiauto.deriveEncoder
+import io.circe.{Encoder, Json}
 import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
-import io.confluent.kafka.serializers.{AbstractKafkaAvroDeserializer}
-import org.apache.avro.AvroRuntimeException
+import io.confluent.kafka.serializers.AbstractKafkaAvroDeserializer
+import org.apache.avro.{AvroRuntimeException, Schema}
 import org.apache.avro.generic.GenericDatumWriter
-import org.apache.avro.io.EncoderFactory
+import org.apache.avro.io.{DatumWriter, EncoderFactory}
+import org.apache.avro.specific.SpecificDatumWriter
+import org.apache.kafka.clients.consumer.ConsumerRecord
 import org.apache.kafka.common.errors.SerializationException
+import pl.touk.nussknacker.engine.kafka.ConsumerRecordUtils
+import pl.touk.nussknacker.engine.kafka.consumerrecord.SerializableConsumerRecord
```

Comment:
I wonder if after changes with `RecordFormatterFactory` it would be possible in avro case to use just `ConsumerRecordToJsonFormatter` and only has some avro specific logic in the factory?

Reply:
I like the idea, will try that with other changes to Formatter that are moved to separate PR.